frame/balances/src/tests.rs

```diff
@@ -660,5 +660,46 @@ macro_rules! decl_tests {
 					assert_eq!(Balances::reserved_balance(1), 0);
 				});
 		}
+
+		#[test]
+		fn set_balance_doesnt_overflow() {
+			<$ext_builder>::default()
+				.existential_deposit(100)
+				.build()
+				.execute_with(|| {
+					assert_ok!(Balances::set_balance(RawOrigin::Root.into(), 1, u64::max_value(), u64::max_value()));
```

Comment:
Have you checked that this test is failing against previous implementation in release mode (like our CI runs it)? Overflows are not not panics in release mode.

Reply:
I would assume in release mode this would not panic, but why is that relevant?

Reply:
Yes, imo it should fail for incorrect implementation and pass for correct one

Reply:
Updated so that tests would fail in release mode using old logic